src/Chapter3.hs

```diff
@@ -971,6 +1070,33 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day 
+  = Monday
+  | Tuesday
+  | Wednesday 
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Eq, Ord, Enum)
+
+class Days a where
+    isWeekend :: a -> Bool
+    nextDay :: a -> Day
+    daysToParty :: a -> Int
+
+instance Days Day where
+    isWeekend :: Day -> Bool
+    isWeekend x = (x == Sunday || x == Saturday)
+
+    nextDay :: Day -> Day
+    -- using `succ`
+    nextDay x = if x == Sunday then Monday else succ x 
+
+    daysToParty :: Day -> Int
+    daysToParty x = if (nextDay x) == Friday then 1 else 1 + daysToParty (nextDay x)
```

Comment:
Your function is recursive, and it works great :+1: 
But you can make it more efficient by using `fromEnum` method of the `Enum` typeclass and the `mod` function